department-of-veterans-affairs/va.gov-cms
# VAGOV-1955 - Workflow Participant notifications

## What this does
Adds `Workflow participants` tab and form to nodes for assigning users that will track workflow changes.
Sends email alerts to users who have been added to node workflow participants forms.
Alerts are only sent if a workflow state change has occurred, and this is triggered by node save.

## Testing Performed
 - Visual
 - Xdebug Profiling
 - Email sending via SMTP (locally)
 - Full functionality test performed on testing instance: http://internal-test-vagovcms-3879-1510337173.us-gov-west-1.elb.amazonaws.com/ 

## Screenshots
![workflow-form](https://user-images.githubusercontent.com/2404547/62426469-43898d00-b6b3-11e9-9ad4-8ac8d1206d12.png)
![confirmation](https://user-images.githubusercontent.com/2404547/62426470-471d1400-b6b3-11e9-9c61-a397686dd78e.png)
![email](https://user-images.githubusercontent.com/2404547/62426472-48e6d780-b6b3-11e9-9444-ae21cd681119.png)

## Definition of done
 - Workflow participants form appears here: `node/{NID}/workflow-participants`
 - Adding a user and saving is successful.
 - Changing a workflow status sends workflow participant an email like the one in the screenshot above.

## To test
 - As an admin, edit the email of the account you are using for testing to an email account you check.
 - As an admin, add the above account to any published node workflow participant form, here:  `node/{NID}/workflow-participants`
 - As an admin, change the workflow state of the above node, and save.
 - Visually verify that you receive an email that is similar in appearance to the screenshot, above.
